### PR TITLE
Add support for static library support

### LINKED ISSUE #232: [FEATURE] Add ability to package all-in-one static library
Add ability to package all bpftime static library archives into a single archive.

Reference: https://github.com/WasmEdge/WasmEdge/blob/f5cf26c66e4bbb9bca00497fce5c814aac7d56fa/lib/api/CMakeLists.txt#L39

Hint: 
- use ${CMAKE_AR} to unarchive each individual to temporary directory, then use ar to repack them into a single archive
- libbpf and spdlog and other dependencies are also needed

### PR DESCRIPTION
**Pull request by hp77-creator**
## Description

Fixes #232 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- [ ] Run build with `BPFTIME_BUILD_STATIC_LIB`

**Test Configuration**:

- Firmware version: 10151.101.3
- Hardware: Macbook Air M2 chip
- Toolchain: Xcode
- SDK: 

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by Officeyutong** [comment]
@hp77-creator How's this issue going on now?

**Comment by hp77-creator** [comment]
@Officeyutong will work on this today, was stuck in some work. will also share the issue that I am facing for local testing

**Comment by hp77-creator** [comment]
@Officeyutong wanted to know how can I test it locally? Need help in that

**Comment by Officeyutong** [comment]
> @Officeyutong wanted to know how can I test it locally? Need help in that

- Make it able to produce a static library
- Write some simple programs that uses bpftime (such as a program that dumps things from shared memory), link it with the static library above and check if it works

**Comment by Officeyutong** [comment]
> Also I am confused by two things
> 
> 1. There was a requirement from @yunwei37 in #231 about using bpftime as a library, what is the scope for that library? Should archive be created for `vm`, `runtime`, `attach`, `daemon` as a one library or something else, need clarity on this.
> 
> 2. If I generate archives how do I test it(already pointed in above message)
> 
> 
> 
> Also, I have tried using an archive helper taking reference from wasmedge but is that necessary or should I change it anywhere?

For `1`, we should package all targets into a single static library

**Comment by yunwei37** [comment]
1. The library should only include runtimes and attach targets, it doesn't include daemon or syscall-server. If users wants to embed userspace eBPF runtime in it's applications like eBPF in kernel, they can use this library.

File: CMakeLists.txt [hunk 1]
```diff
@@ -117,8 +117,12 @@ add_subdirectory(benchmark)
 if (${BPFTIME_BUILD_STATIC_LIB}) 
 message(STATUS " Adding libraries to static archive")
 bpftime_add_static_lib_component_command(runtime)
-bpftime_add_static_lib_component_command(attach)
-bpftime_add_static_lib_component_command(vm)
+bpftime_add_static_lib_component_command(base_attach_impl)
+bpftime_add_static_lib_component_command(frida_uprobe_attach_impl)
+bpftime_add_static_lib_component_command(syscall_trace_attach_impl)
+bpftime_add_static_lib_component_command(text_segment_transformer)
```
Comment (yunwei37) [inline CMakeLists.txt:123]: I think we may not need to package text_segment_transformer into the static target?

Comment (hp77-creator) [reply CMakeLists.txt:123]: yea you are right, It was generating a shared library, I was getting an error when I tried to package it using `ar`, Also, I wll create a new PR, closing this one, cuz it is a lot behind the master branch, will create a new branch from the master and send PR from that.

**Comment by hp77-creator** [comment]
Closing this one because I have created another #294 PR for same changes.

**Pull request closed without merging**